src/index.js

```diff
@@ -1,44 +1,116 @@
+  localStorage.setItem('tasks', itemsJson);
+}
 
 const postContainer = document.querySelector('.tasklist');
+const addButton = document.querySelector('.newtask-return');
+const items = getItems();
+
+function updateItem(index, key, value) {
+  items[index][key] = value;
+  setItems(items);
+}
+
+function deleteItem(index) {
+  items.splice(index, 1);
+  setItems(items);
+  postContainer.dispatchEvent(new CustomEvent('itemDeleted'));
+}
+
+const createItemElement = (item, index) => {
+  const itemElement = document.createElement('div');
+  itemElement.classList.add('task-group');
+  itemElement.innerHTML = `
+    <input class="task-item" type="checkbox">
+    <input class="task-item-text" placeholder="Task item...">
+  `;
+
+  const myIcon = new Image();
+  myIcon.src = More;
+  myIcon.classList.add('task-item-more');
+  itemElement.appendChild(myIcon);
+
+  const myDelete = new Image();
+  myDelete.src = Delete;
+  myDelete.classList.add('task-item-delete');
+  itemElement.appendChild(myDelete);
+
+  const descriptionInput = itemElement.querySelector('.task-item-text');
+  const completedInput = itemElement.querySelector('.task-item');
+  const deleteButton = itemElement.querySelector('.task-item-delete');
+
+  descriptionInput.value = item.description;
+  completedInput.checked = item.completed;
+
+  descriptionInput.addEventListener('change', () => {
+    const newValue = descriptionInput.value;
+    item.description = newValue;
+    updateItem(index, 'description', newValue);
+    itemElement.dispatchEvent(new CustomEvent('itemChanged'));
+  });
+
+  completedInput.addEventListener('change', () => {
+    const newValue = completedInput.checked;
+    item.completed = newValue;
+    updateItem(index, 'completed', newValue);
+    itemElement.dispatchEvent(new CustomEvent('itemChanged'));
+  });
 
-const postMethods = () => {
-  // Sort the cardData array by the index value in ascending order
-  const sortedData = _.sortBy(cardData, [(data) => parseInt(data.index, 10)]);
-
-  sortedData.forEach((postData) => {
-    const postElement = document.createElement('div');
-    postElement.classList.add('task-group');
-    postElement.innerHTML = `
-        <input id="${postData.index}" type="checkbox" name="task-item" value="${postData.index}"> <input id="input-${postData.index}" class="task-item-text" placeholder="${postData.description}">
-      `;
-    const myIcon = new Image();
-    myIcon.src = More;
-    myIcon.classList.add('task-item-more');
-    postElement.appendChild(myIcon);
-
-    postContainer.appendChild(postElement);
+  deleteButton.addEventListener('click', () => {
+    deleteItem(index);
   });
+
+  return itemElement;
 };
 
-postMethods();
\ No newline at end of file
+function refreshList() {
+  postContainer.innerHTML = '';
+
+  items.forEach((item, index) => {
+    const itemElement = createItemElement(item, index);
+    postContainer.appendChild(itemElement);
+  });
+}
+
+function addItem() {
```

Comment:
- [ ] Kindly 🙏🏻 follow the best practices for the `JavaScript` by using the `arrow` functions or ES6 style JavaScript code instead of `ES5` functions because ES6 syntax makes your code more readable, manageable and clean. Even the functionalities provided in the `ES6`, results are more predictable and reliable. Please fix the things as I mentioned.